testsuite/Makefile

```diff
@@ -178,10 +184,66 @@ clean:
 	  (cd `dirname $$file` && $(MAKE) BASEDIR=$(BASEDIR) clean); \
 	done
 
+# In order to make this grep expression readable, spaces end up being introduced
+# so a small make-hoop is jumped through to remove spaces from it when it's
+# used.
+SKIP_LOGIC=[{,]\( \
+  system-$(SYSTEM)\|arch-$(ARCH) \
+  $(if $(filter win32,$(UNIX_OR_WIN32)),\|win32) \
+  $(if $(filter unix,$(OTHERLIBRIARIES)),,\|!unix) \
+  $(if $(FORTRAN_COMPILER),,\|!fortran) \
+  $(if $(filter true,$(BYTECODE_ONLY)),\|bytecode) \
+  $(if $(filter false,$(FLAMBDA)),\|!flambda) \
+  $(if $(shell which csc.exe 2>/dev/null),,\|!csharp) \
+  \|!system-[^,}]*\) \
+  [,}]
```

Comment:
@damiendoligez is intended to appreciate my Herculean efforts to keep the lines below 80 chars.........